pkg/loop/internal/core/services/capability/standard_capability.go

```diff
@@ -0,0 +1,223 @@
+package capability
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	capabilitiespb "github.com/smartcontractkit/chainlink-common/pkg/capabilities/pb"
+	"github.com/smartcontractkit/chainlink-common/pkg/loop/internal/core/services/errorlog"
+	"github.com/smartcontractkit/chainlink-common/pkg/loop/internal/core/services/keyvalue"
+	"github.com/smartcontractkit/chainlink-common/pkg/loop/internal/core/services/pipeline"
+	"github.com/smartcontractkit/chainlink-common/pkg/loop/internal/core/services/telemetry"
+	"github.com/smartcontractkit/chainlink-common/pkg/loop/internal/goplugin"
+	"github.com/smartcontractkit/chainlink-common/pkg/loop/internal/net"
+	"github.com/smartcontractkit/chainlink-common/pkg/loop/internal/pb"
+	relayersetpb "github.com/smartcontractkit/chainlink-common/pkg/loop/internal/pb/relayerset"
+	"github.com/smartcontractkit/chainlink-common/pkg/loop/internal/relayerset"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/core"
+)
+
+type StandardCapability interface {
+	services.Service
+	capabilities.BaseCapability
+	Initialise(ctx context.Context, config string, telemetryService core.TelemetryService, store core.KeyValueStore,
+		capabilityRegistry core.CapabilitiesRegistry, errorLog core.ErrorLog,
+		pipelineRunner core.PipelineRunnerService, relayerSet core.RelayerSet) error
+}
+
+type StandardCapabilityClient struct {
+	*goplugin.PluginClient
+	capabilitiespb.StandardCapabilityClient
+	*baseCapabilityClient
+	*goplugin.ServiceClient
+	*net.BrokerExt
+
+	resources []net.Resource
+}
+
+var _ StandardCapability = (*StandardCapabilityClient)(nil)
+
+func NewStandardCapabilityClient(brokerExt *net.BrokerExt, conn *grpc.ClientConn) *StandardCapabilityClient {
+	return &StandardCapabilityClient{
+		PluginClient:             goplugin.NewPluginClient(brokerExt.Broker, brokerExt.BrokerConfig, conn),
+		ServiceClient:            goplugin.NewServiceClient(brokerExt, conn),
+		StandardCapabilityClient: capabilitiespb.NewStandardCapabilityClient(conn),
+		baseCapabilityClient:     newBaseCapabilityClient(brokerExt, conn),
+		BrokerExt:                brokerExt,
+	}
+}
+
+func (c *StandardCapabilityClient) Initialise(ctx context.Context, config string, telemetryService core.TelemetryService,
+	keyValueStore core.KeyValueStore, capabilitiesRegistry core.CapabilitiesRegistry, errorLog core.ErrorLog,
+	pipelineRunner core.PipelineRunnerService, relayerSet core.RelayerSet) error {
+	telemetryID, telemetryRes, err := c.ServeNew("Telemetry", func(s *grpc.Server) {
+		pb.RegisterTelemetryServer(s, telemetry.NewTelemetryServer(telemetryService))
+	})
+	if err != nil {
+		return fmt.Errorf("failed to serve new telemetry: %w", err)
+	}
+	c.resources = append(c.resources, telemetryRes)
```

Comment:
are we leaking resources if any of the ServeNew fail? do we need a defer that `Close`s all the resources if there an initialization error?

Reply:
yes, added some code to cleanup resources in the event of error.